runtime/src/iree/hal/drivers/amdgpu/util/libhsa.c

```diff
@@ -0,0 +1,692 @@
+    case HSA_STATUS_ERROR_INVALID_WAVEFRONT:
+      return "HSA_STATUS_ERROR_INVALID_WAVEFRONT";
+    case HSA_STATUS_ERROR_INVALID_SIGNAL_GROUP:
+      return "HSA_STATUS_ERROR_INVALID_SIGNAL_GROUP";
+    case HSA_STATUS_ERROR_INVALID_RUNTIME_STATE:
+      return "HSA_STATUS_ERROR_INVALID_RUNTIME_STATE";
+    case HSA_STATUS_ERROR_FATAL:
+      return "HSA_STATUS_ERROR_FATAL";
+  }
+}
+
+// Returns a string literal describing the given HSA status code.
+// We inline the string mapping here instead of calling hsa_status_string so
+// that we can customize the error messages to our uses (and avoid the need for
+// carrying the libhsa pointer to resolve statuses).
+static const char* iree_hsa_status_description(hsa_status_t status) {
+  switch (status) {
+    default:
+      return "Unknown.";
+    case HSA_STATUS_SUCCESS:
+      return "The function has been executed successfully.";
+    case HSA_STATUS_INFO_BREAK:
+      return "A traversal over a list of elements has been interrupted by the "
+             "application before completing.";
+    case HSA_STATUS_ERROR:
+      return "A generic error has occurred.";
+    case HSA_STATUS_ERROR_INVALID_ARGUMENT:
+      return "One of the actual arguments does not meet a precondition stated "
+             "in the documentation of the corresponding formal argument.";
+    case HSA_STATUS_ERROR_INVALID_QUEUE_CREATION:
+      return "The requested queue creation is not valid.";
+    case HSA_STATUS_ERROR_INVALID_ALLOCATION:
+      return "The requested allocation is not valid.";
+    case HSA_STATUS_ERROR_INVALID_AGENT:
+      return "The agent is invalid.";
+    case HSA_STATUS_ERROR_INVALID_REGION:
+      return "The memory region is invalid.";
+    case HSA_STATUS_ERROR_INVALID_SIGNAL:
+      return "The signal is invalid.";
+    case HSA_STATUS_ERROR_INVALID_QUEUE:
+      return "The queue is invalid.";
+    case HSA_STATUS_ERROR_OUT_OF_RESOURCES:
+      return "The HSA runtime failed to allocate the necessary resources. This "
+             "error may also occur when the HSA runtime needs to spawn threads "
+             "or create internal OS-specific events.";
+    case HSA_STATUS_ERROR_INVALID_PACKET_FORMAT:
+      return "The AQL packet is malformed.";
+    case HSA_STATUS_ERROR_RESOURCE_FREE:
+      return "An error has been detected while releasing a resource.";
+    case HSA_STATUS_ERROR_NOT_INITIALIZED:
+      return "An API other than hsa_init has been invoked while the reference "
+             "count of the HSA runtime is 0.";
+    case HSA_STATUS_ERROR_REFCOUNT_OVERFLOW:
+      return "The maximum reference count for the object has been reached.";
+    case HSA_STATUS_ERROR_INCOMPATIBLE_ARGUMENTS:
+      return "The arguments passed to a functions are not compatible.";
+    case HSA_STATUS_ERROR_INVALID_INDEX:
+      return "The index is invalid.";
+    case HSA_STATUS_ERROR_INVALID_ISA:
+      return "The instruction set architecture is invalid.";
```

Comment:
For these sorts of status descriptions, are we also including the values (e.g. `index`, `isa`) that were invalid in what gets logged when an error occurs?